### PR TITLE
FramingIO optimizations

### PR DESCRIPTION
**Pull request by pablogs9**

### CODE REVIEW

**Comment by pablogs9** [comment]
Related https://github.com/eProsima/Micro-XRCE-DDS-Client/pull/267

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Agent%20Manual%20Linux/502/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Agent%20Manual%20Windows/503/)

File: src/cpp/transport/stream_framing/StreamFramingProtocol.cpp [hunk 1]
```diff
@@ -255,7 +274,7 @@ size_t FramingIO::read_framed_msg(
                         {
                             state_ = InputState::UXR_FRAMING_READING_SRC_ADDR;
                         }
-                        else if (0 < transport_read(timeout, transport_rc))
+                        else if (0 < transport_read(timeout, transport_rc, msg_len_ - msg_pos_))
```
Comment (Acuadros95) [inline src/cpp/transport/stream_framing/StreamFramingProtocol.cpp:277]:
```suggestion
                        else if (0 < transport_read(timeout, transport_rc, (msg_len_ - msg_pos_) + 2))
```
I think we should include the CRC read here

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Agent%20Manual%20Linux/506/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Agent%20Manual%20Windows/507/)

File: src/cpp/transport/stream_framing/StreamFramingProtocol.cpp (hunk 1 above)
Comment (Acuadros95) [reply src/cpp/transport/stream_framing/StreamFramingProtocol.cpp]: Done https://github.com/eProsima/Micro-XRCE-DDS-Client/pull/267/commits/937ea81d38832563e03a6d95c6986d322bc5b8b1

**Comment by Acuadros95** [comment]
Tested and looking good with crazyflie demo

**Review by Acuadros95 — APPROVED** [review]

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Agent%20Manual%20Linux/507/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Agent%20Manual%20Windows/508/)

**Pull request merged**